cmd/spire-agent/cli/run/run.go

```diff
@@ -400,6 +402,17 @@ func NewAgentConfig(c *Config, logOptions []log.Option, allowUnknownConfig bool)
 		}
 	}
 
+	if c.Agent.Experimental.MaxSvidCacheSize != 0 {
```

Comment:
```suggestion
	if c.Agent.Experimental.MaxSvidCacheSize > 0 {
```

Reply:
and then unit test negative value rejection probably?

maybe negative values should be logged and/or error out since it means operator has misconfigured

Reply:
added check and unit tests for negative values.